lib/mongo/cached_cursor.rb

```diff
@@ -0,0 +1,60 @@
+# A Cursor that attempts to load documents from memory first before hitting
+# the database if the same query has already been executed.
+#
+# @since 5.0.0
+
+module Mongo
+  class CachedCursor < Cursor
+
+    # We iterate over the cached documents if they exist already in the
+    # cursor otherwise proceed as normal.
+    #
+    # @example Iterate over the documents.
+    #   cursor.each do |doc|
+    #     # ...
+    #   end
+    #
+    # @since 5.0.0
```

Comment:
All of the since tags should be removed since they reference Mongoid versions (and we aren't using them anymore in new code).